File: c++/Easy/Part002/Top Scoring batsman Name.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <limits>

struct Batsman {
    std::string name;
    int runs;
};

int main() {
    int n;
    std::cin >> n;
    std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n'); // Ignore the newline character after n

    std::vector<Batsman> batsmen(n);

    for (int i = 0; i < n; ++i) {
        std::string input;
        std::getline(std::cin, input);

        std::stringstream ss(input);
        std::string name;
        std::string runs_str;

        std::getline(ss, name, ',');
        std::getline(ss, runs_str);

        int runs = std::stoi(runs_str);

        batsmen[i] = {name, runs};
    }

    Batsman topScorer = batsmen[0];
    for (int i = 1; i < n; ++i) {
        if (batsmen[i].runs > topScorer.runs) {
            topScorer = batsmen[i];
        }
    }

    std::cout << topScorer.name << std::endl;

    return 0;
}

File: Java/Easy/Part002/Top Scoring Batsman Name.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;

class Batsman {
    String name;
    int runs;

    Batsman(String name, int runs) {
        this.name = name;
        this.runs = runs;
    }
}

public class Hello {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read the number of batsmen
        int n = scanner.nextInt();
        scanner.nextLine(); // Ignore the newline character after n
        
        List<Batsman> batsmen = new ArrayList<>(n);
        
        for (int i = 0; i < n; ++i) {
            String input = scanner.nextLine();
            String[] parts = input.split(",");
            String name = parts[0].trim();
            int runs = Integer.parseInt(parts[1].trim());
            batsmen.add(new Batsman(name, runs));
        }
        
        Batsman topScorer = batsmen.get(0);
        for (int i = 1; i < n; ++i) {
            if (batsmen.get(i).runs > topScorer.runs) {
                topScorer = batsmen.get(i);
            }
        }
        
        System.out.println(topScorer.name);
        
        scanner.close();
    }
}
